test/database.test.js

```diff
@@ -0,0 +1,62 @@
+import {
+  afterAll,
+  beforeAll, beforeEach, describe, expect, it,
+} from 'vitest';
+import { client } from '../connections.js';
+import {
+  addNewSoldier, lookForSoldier, lookForAllSoldiers, dbName, dbCollection,
+} from '../database.js';
+
+beforeAll(async () => {
+  await client.connect();
+});
+
+afterAll(async () => {
+  await client.close();
+});
+
+beforeEach(async () => {
+  await client.db(dbName).collection(dbCollection).deleteMany({});
+});
+
+describe('addNewSoldier function', () => {
+  it('Should insert soldier', async () => {
+    const res = await addNewSoldier({ name: 'Amit' });
+    expect(res.acknowledged).eq(true);
+  });
+});
+
+const testId = {
+  _id: '9033543',
+};
+
+const TestSoldier1 = {
+  _id: '12345',
+  name: 'Hili',
+  rank: '1',
+  limitations: ['none', 'none'],
+};
+
+const TestSoldier2 = {
+  _id: '45678',
+  name: 'Hili',
+  rank: '1',
+  limitations: ['none', 'none'],
+};
+
+describe('lookForSoldier function', () => {
+  it('should find correct soldier', async () => {
+    await addNewSoldier(testId);
+    const res = await lookForSoldier(testId);
+    expect(res._id).eq('9033543');
```

Comment:
```suggestion
    expect(res._id).eq(testId._id);
```

or if `testId` is `string`

```suggestion
    expect(res._id).eq(testId);
```

Reply:
fixed